test/hotspot/jtreg/compiler/loopopts/TestLoopLimitNodeElimination.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8260370
+ * @summary C2: LoopLimit node is not eliminated
+ *
+ * @run main/othervm
+ *      -Xcomp
+ *      -XX:CompileOnly=compiler/loopopts/TestLoopLimitNodeElimination
+ *      compiler.loopopts.TestLoopLimitNodeElimination
+ */
+
+package compiler.loopopts;
+
+public class TestLoopLimitNodeElimination {
+    int a = 400;
+    static int counter = 0;
+    long b[] = new long[a];
+
+    void c(String[] d) {
```

Comment:
Please, don't use single latter for constants and method names. It is better to not use them for anything - all names should be meaningful.

Also you need to change RFE to Bug. the process only allows test bugs:
http://openjdk.java.net/jeps/3#rdp-2
`RDP2: P1–P5 doc/test bugs`

Reply:
Vladimir, there is nothing meaningful about these constants, methods, variables — the whole test is randomly generated.

Reply:
Got it. But even then, you can at least rename 'c()' to 'test()' and 'a' to 'N'. And add comment that it is randomly generated to avoid complains later.